plugin/entrySchema.go

```diff
@@ -66,13 +66,14 @@ func schema(e Entry) (*EntrySchema, error) {
 }
 
 type entrySchema struct {
-	Label               string      `json:"label"`
-	Description         string      `json:"description,omitempty"`
-	Singleton           bool        `json:"singleton"`
-	Actions             []string    `json:"actions"`
-	MetaAttributeSchema *JSONSchema `json:"meta_attribute_schema"`
-	MetadataSchema      *JSONSchema `json:"metadata_schema"`
-	Children            []string    `json:"children"`
+	Label               string            `json:"label"`
+	Description         string            `json:"description,omitempty"`
+	Singleton           bool              `json:"singleton"`
+	Signals             map[string]string `json:"signals,omitempty"`
```

Comment:
I also considered supporting a tuple for methods like `(<method_name>, <method_schema>)` but that required a bit of a refactor and didn't generalize well to methods like `read`, `exec`, `schema` that didn't need `<method_schema>`. This way's also symmetric with wash ruby where there's a separate `signals` helper that lets you set the supported signals.

Reply:
Hmm, it looks like for Docker containers you can also send any one of the supported Linux signals (like `SIGHUP`, `SIGSTOP`, etc.) via `docker kill`. It'd be tedious for a user to have to specify these signals in the schema. Wonder if we could provide a short-hand for that.

Since these can be vendor-specific, I think we should change `Signals` to `map[signals]SignalSchema`

Reply:
Thinking about something like `map[string]SignalSchema` where `SignalSchema` is something like
```
type SignalSchema struct {
   Description
   Regex
}
```

If a Regex is provided, then we'll validate the provided signal using that. (At a higher-level, SignalSchemas w/ a Regex represent the schema of a group of signals, while w/o a Regex we are talking about a single Signal). Otherwise, we'll go w/ the signal name. This way, I can specify OS-specific signals as
```
"linux" => {
  "Description" => "Linux signals",
  "Regex" => "SIG*"
}
```

I'd prefer something like this instead of a hard-coded "linux" constant that people could use b/c that preserves Signal's general nature.

Could also change `Regex` to `Signals` to make validation easier.

Reply:
Yeah, I think storing this as an array to preserve the order makes sense.

Linux signals introduces the more general problem of "My API has a bunch of signals, many of which fit a specific group/category. I'd like to have a way of telling Wash that information without having to list every signal because not only is that tedious, but it will also annoy users of my plugin, especially since those users will (likely) be aware of what that specific group/category of signals represent when they view my entry's set of supported signals."

Having a Linux constant isn't a good solution to that problem, esp. when the general solution is (probably) not that much harder.

Not sure what you mean by extra flag though. What would that look like?

Reply:
Ah. I think I'd missed your last points. a Regex seems fine.